src/fielddef.ts

```diff
@@ -147,7 +147,8 @@ export function field(fieldDef: FieldDef, opt: FieldRefOption = {}) {
 
     if (!opt.nofn) {
       if (fieldDef.bin) {
-        fn = 'bin';
+        const bin: Bin = isBoolean(fieldDef.bin) ? {} : {...fieldDef.bin};
```

Comment:
You forgot to handle the case that we automatically add autoCount to bin.  (Extract that part of that to binToString too)